jdbc/src/main/java/io/kestra/jdbc/runner/JdbcExecutor.java

```diff
@@ -1170,27 +1170,34 @@ private void toExecution(Executor executor, boolean ignoreFailure) {
                     // this could only happen for KILLED execution.
                     boolean queuedThenKilled = execution.getState().getCurrent() == State.Type.KILLED
                         && execution.getState().getHistories().stream().anyMatch(h -> h.getState().isQueued())
-                        && execution.getState().getHistories().stream().noneMatch(h -> h.getState().isRunning());
+                        && execution.getState().getHistories().stream().noneMatch(h -> h.getState().onlyRunning());
                     boolean concurrencyShortCircuitState = Concurrency.possibleTransitions(execution.getState().getCurrent())
                         && execution.getState().getHistories().get(execution.getState().getHistories().size() - 2).getState().isCreated();
                     if (!queuedThenKilled && !concurrencyShortCircuitState) {
-                        concurrencyLimitStorage.decrement(executor.getFlow());
-
-                        if (executor.getFlow().getConcurrency().getBehavior() == Concurrency.Behavior.QUEUE) {
-                            var finalFlow = executor.getFlow();
-                            executionQueuedStorage.pop(executor.getFlow().getTenantId(),
-                                executor.getFlow().getNamespace(),
-                                executor.getFlow().getId(),
-                                throwBiConsumer((dslContext, queued) -> {
-                                    var newExecution = queued.withState(State.Type.RUNNING);
-                                    concurrencyLimitStorage.increment(dslContext, finalFlow);
-                                    executionQueue.emit(newExecution);
-                                    metricRegistry.counter(MetricRegistry.METRIC_EXECUTOR_EXECUTION_POPPED_COUNT, MetricRegistry.METRIC_EXECUTOR_EXECUTION_POPPED_COUNT_DESCRIPTION, metricRegistry.tags(newExecution)).increment();
-
-                                    // process flow triggers to allow listening on RUNNING state after a QUEUED state
-                                    processFlowTriggers(newExecution);
-                                })
-                            );
+                        var currentFlow = executor.getFlow();
+
+                        if (!(execution.getState().getCurrent() == State.Type.KILLED && executor.getFrom().contains("addWorkerTaskResult"))) {
+                            concurrencyLimitStorage.decrement(executor.getFlow());
+
+                            int currentConcurencyLimit = this.concurrencyLimitStorage.fetchOne(currentFlow).getRunning();
+
+                            if (executor.getFlow().getConcurrency().getBehavior() == Concurrency.Behavior.QUEUE
+                                && currentConcurencyLimit < currentFlow.getConcurrency().getLimit()) {
```

Comment:
This is not safe if called concurrently so I don't think it should be added here even if I understand the rational